mhkit/wave/resource.py

```diff
@@ -87,26 +86,24 @@ def elevation_spectrum(
             + "temporal spacing for eta."
         )
 
-    S = xr.Dataset()
-    for var in eta.data_vars:
```

Comment:
Users can still input Datasets, but right now all variables must have the same dimensions, it will be converted to a DataArray up front. The function then returns a DataArray. 

I'll look at reinstating a dataset/dataframe loop so that those types are returned

Reply:
I added this loop again. Our previous slowdown of large pandas DataFrames --> xarray datasets could occur in these two functions now, but I don't think that is a typical use case. For example in the case described in #331, it's unlikely a user would have thousands of different wave elevation time series and convert them all to wave spectra (likewise for thousands of distinct spectra being converted to elevation time series). If that case does come up, the slow down should not be due to our implementation but the large quantity of data involved.

Reply:
Right now if a user inputs a multivariate type (Dataset, DataFrame), they will get a multivariate type back, but whether its a Dataset or DataFrame is still controlled by the ``to_pandas`` flag, which is consistent with other functions containing this flag. 

It might be worth making a more complex default for ``to_pandas`` where a use could specify True/False, and if they don't specify the function defaults to return the type (xarray/pandas) that they input. However this should be a separate PR that address all instances of this flag for consistency

Reply:
Apologies. That was an oversite on my part on the `to_pandas` parameter.

I still want to add the above test to the testing suite https://github.com/akeeste/MHKiT-Python/pull/6